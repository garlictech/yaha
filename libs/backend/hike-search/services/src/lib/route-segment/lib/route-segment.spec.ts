import { routeSegmentData } from './test/fixtures';
import { tap } from 'rxjs/operators';

import { validateCoordinatesWithElevation } from '../../graphql-api';

describe('Test route segment library', () => {
  test('Validate a valid route segment coordinate array', done => {
    validateCoordinatesWithElevation(routeSegmentData())
      .pipe(tap(result => expect(result).toMatchSnapshot()))
      .subscribe(() => done());
  });
});
