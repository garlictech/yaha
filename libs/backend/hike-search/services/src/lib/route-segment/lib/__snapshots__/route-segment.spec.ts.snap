// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test route segment library Validate a valid route segment coordinate array 1`] = `
Array [
  Array [
    18.944105,
    47.547293,
    310.14767456,
  ],
  Array [
    18.94408,
    47.547347,
    310.38842773,
  ],
  Array [
    18.94287,
    47.546998,
    319.9692688,
  ],
  Array [
    18.941683,
    47.546695,
    314.39724731,
  ],
  Array [
    18.941622,
    47.54679,
    314.58563232,
  ],
  Array [
    18.941543,
    47.546763,
    314.21194458,
  ],
  Array [
    18.941432,
    47.546842,
    314.27514648,
  ],
  Array [
    18.941384,
    47.546814,
    313.97406006,
  ],
  Array [
    18.941161,
    47.546985,
    314.11685181,
  ],
  Array [
    18.941164,
    47.54692,
    313.80453491,
  ],
  Array [
    18.940654,
    47.546861,
    313.12869263,
  ],
  Array [
    18.940051,
    47.546946,
    312.30062866,
  ],
  Array [
    18.939779,
    47.54695,
    311.80328369,
  ],
  Array [
    18.939548,
    47.54672,
    312.03903198,
  ],
  Array [
    18.939471,
    47.546756,
    311.6595459,
  ],
  Array [
    18.938574,
    47.546014,
    316.38641357,
  ],
  Array [
    18.938603,
    47.545895,
    317.69711304,
  ],
  Array [
    18.938547,
    47.545731,
    319.48638916,
  ],
  Array [
    18.938432,
    47.545568,
    321.07003784,
  ],
  Array [
    18.938094,
    47.545305,
    324.22280884,
  ],
  Array [
    18.937578,
    47.544803,
    331.98623657,
  ],
  Array [
    18.936695,
    47.544105,
    338.59191895,
  ],
  Array [
    18.936056,
    47.543515,
    338.39709473,
  ],
  Array [
    18.935922,
    47.543489,
    337.64285278,
  ],
  Array [
    18.935815,
    47.543406,
    337.23001099,
  ],
  Array [
    18.935475,
    47.543392,
    336.17849731,
  ],
  Array [
    18.935054,
    47.543319,
    336.77658081,
  ],
  Array [
    18.934455,
    47.543152,
    338.78091431,
  ],
  Array [
    18.933758,
    47.543081,
    340.22817993,
  ],
  Array [
    18.932674,
    47.543191,
    343.54614258,
  ],
  Array [
    18.93227,
    47.543209,
    344.85687256,
  ],
  Array [
    18.931864,
    47.543295,
    347.18695068,
  ],
  Array [
    18.930975,
    47.543336,
    350.5112915,
  ],
  Array [
    18.9307,
    47.543317,
    350.74667358,
  ],
  Array [
    18.930383,
    47.543333,
    350.33978271,
  ],
  Array [
    18.930149,
    47.54337,
    349.92706299,
  ],
  Array [
    18.929933,
    47.543426,
    349.72317505,
  ],
  Array [
    18.929674,
    47.543526,
    349.43191528,
  ],
  Array [
    18.929488,
    47.543636,
    349.11151123,
  ],
  Array [
    18.92894,
    47.5438,
    347.91241455,
  ],
  Array [
    18.928493,
    47.544242,
    344.87994385,
  ],
  Array [
    18.926887,
    47.545307,
    342.10174561,
  ],
  Array [
    18.926267,
    47.545639,
    345.08648682,
  ],
  Array [
    18.924378,
    47.546453,
    357.7701416,
  ],
  Array [
    18.924225,
    47.546483,
    358.87445068,
  ],
  Array [
    18.923394,
    47.5465,
    364.09164429,
  ],
  Array [
    18.923133,
    47.546517,
    365.88812256,
  ],
  Array [
    18.922597,
    47.546587,
    370.05331421,
  ],
  Array [
    18.921678,
    47.546674,
    380.14584351,
  ],
  Array [
    18.921472,
    47.546592,
    381.44055176,
  ],
  Array [
    18.921207,
    47.54638,
    382.81756592,
  ],
  Array [
    18.920358,
    47.546133,
    387.75637817,
  ],
  Array [
    18.919816,
    47.546007,
    390.14459229,
  ],
  Array [
    18.91923,
    47.545937,
    392.28833008,
  ],
  Array [
    18.918398,
    47.546064,
    393.94714355,
  ],
  Array [
    18.918097,
    47.546144,
    394.90817261,
  ],
  Array [
    18.917434,
    47.546244,
    396.61340332,
  ],
  Array [
    18.917253,
    47.546314,
    397.50057983,
  ],
  Array [
    18.916902,
    47.546559,
    400.67974854,
  ],
  Array [
    18.916311,
    47.546879,
    405.03366089,
  ],
  Array [
    18.916206,
    47.546947,
    405.65313721,
  ],
  Array [
    18.916016,
    47.547108,
    407.20880127,
  ],
  Array [
    18.915228,
    47.547407,
    410.38626099,
  ],
  Array [
    18.914593,
    47.547618,
    413.46881104,
  ],
  Array [
    18.91398,
    47.547874,
    417.44528198,
  ],
  Array [
    18.912907,
    47.548059,
    419.4005127,
  ],
  Array [
    18.911989,
    47.548178,
    418.88354492,
  ],
  Array [
    18.911221,
    47.54825,
    419.21261597,
  ],
  Array [
    18.909783,
    47.548591,
    422.58114624,
  ],
  Array [
    18.909469,
    47.548647,
    422.42810059,
  ],
  Array [
    18.90922,
    47.548669,
    421.99728394,
  ],
  Array [
    18.908881,
    47.548645,
    420.48669434,
  ],
  Array [
    18.90836,
    47.548674,
    419.74649048,
  ],
  Array [
    18.907442,
    47.548746,
    421.22427368,
  ],
  Array [
    18.906574,
    47.548874,
    425.45825195,
  ],
  Array [
    18.905662,
    47.548915,
    427.27288818,
  ],
  Array [
    18.904999,
    47.549064,
    429.07095337,
  ],
  Array [
    18.904474,
    47.549155,
    429.80184937,
  ],
  Array [
    18.902029,
    47.549859,
    441.16436768,
  ],
  Array [
    18.901571,
    47.550048,
    444.38104248,
  ],
  Array [
    18.899512,
    47.5506,
    451.74633789,
  ],
  Array [
    18.898272,
    47.551016,
    458.39077759,
  ],
  Array [
    18.89708,
    47.55116,
    457.80801392,
  ],
  Array [
    18.895266,
    47.551694,
    464.21292114,
  ],
  Array [
    18.895183,
    47.551776,
    465.78579712,
  ],
  Array [
    18.892257,
    47.552454,
    473.26721191,
  ],
  Array [
    18.892091,
    47.552526,
    474.28665161,
  ],
  Array [
    18.891916,
    47.552624,
    475.49542236,
  ],
  Array [
    18.891599,
    47.552838,
    478.74636841,
  ],
  Array [
    18.891514,
    47.553026,
    481.94418335,
  ],
  Array [
    18.891438,
    47.553078,
    482.61410522,
  ],
  Array [
    18.891328,
    47.55319,
    484.21609497,
  ],
  Array [
    18.891178,
    47.553278,
    485.35205078,
  ],
  Array [
    18.890794,
    47.553441,
    486.4262085,
  ],
  Array [
    18.890263,
    47.553617,
    486.61288452,
  ],
  Array [
    18.88974,
    47.553828,
    489.05419922,
  ],
  Array [
    18.88934,
    47.554014,
    491.88522339,
  ],
  Array [
    18.888692,
    47.554235,
    495.26959229,
  ],
  Array [
    18.888479,
    47.554341,
    496.57012939,
  ],
  Array [
    18.888329,
    47.554438,
    497.52890015,
  ],
  Array [
    18.887979,
    47.55477,
    502.11694336,
  ],
  Array [
    18.887586,
    47.555073,
    507.6736145,
  ],
  Array [
    18.886912,
    47.555325,
    512.60772705,
  ],
  Array [
    18.886488,
    47.555407,
    513.85552979,
  ],
  Array [
    18.885684,
    47.555627,
    517.58709717,
  ],
  Array [
    18.885144,
    47.555692,
    518.3883667,
  ],
  Array [
    18.884362,
    47.555986,
    520.69274902,
  ],
  Array [
    18.884112,
    47.5561,
    521.90924072,
  ],
  Array [
    18.884015,
    47.556231,
    523.74584961,
  ],
  Array [
    18.883406,
    47.556591,
    525.92462158,
  ],
  Array [
    18.882882,
    47.556852,
    527.60302734,
  ],
]
`;
